packages/common/src/utils/metricsExplorer.ts

```diff
@@ -185,20 +191,31 @@ export const getMetricExplorerDataPointsWithCompare = (
 
     const offsetGroupByCompareMetricRows = mapKeys(
         groupByCompareMetricRows,
-        (_, date) => offsetFunction(new Date(date)).toString(),
+        (_, date) =>
+            comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD
+                ? oneYearForward(new Date(date)).toString()
+                : date,
     );
 
     const dates = new Set([
         ...Object.keys(groupByMetricRows),
         ...Object.keys(offsetGroupByCompareMetricRows),
     ]);
 
+    const compareMetricId =
+        comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD
+            ? metricId
+            : getItemId({
+                  table: comparison.metricTable,
+                  name: comparison.metricName,
+              });
```

Comment:
maybe we can be more explicit with the enum for this case? so it's not the _default_ if we add more 

can be part 2

Reply:
+